python/ray/tune/trial_runner.py

```diff
@@ -234,13 +235,16 @@ def _process_events(self):
                 result, terminate=(decision == TrialScheduler.STOP))
 
             if decision == TrialScheduler.CONTINUE:
-                if trial.should_checkpoint():
+                if trial.should_checkpoint(result):
```

Comment:
We can actually just take this out of the if-else right? 
```
if trial.should_checkpoint(result):
    self.trial_executor.save(trial)

if decision == TrialScheduler.CONTINUE:
    self.trial_executor.continue_training(trial)
elif decision == ...
```

Reply:
Yeah! That makes it cleaner. I have pushed the changed. I left the comment with the TODO (blocking call to save) for you untouched :D